microscopic-monks/primal/gui/inventory.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+
+from kivy.graphics.instructions import RenderContext
+
+from primal.engine.sprite import Sprite, ColorSprite, Text
+
+import json
+
+
+class Inventory:
+    def __init__(self, pos):
+        self.item_data = None
+        self.inventory_data = [[]]
+
+        self.items = []
+        self.grid = []
+        self.amounts = []
+        self.active = 0
+
+        for i in range(10):
+            self.items.append(Sprite('blank.png', (pos[0] + 5, pos[1] + 60 * i + 5), (40, 40)))
+
+        for i in range(10):
+            self.grid.append(
+                ColorSprite(None, (pos[0], pos[1] + 60 * i), (50, 50), (1, 1, 1, .25)))
+
+        for i in range(10):
+            label = Text('16', (pos[0] + 2, pos[1] + 60 * i), 15)
+            label.set_color((0, 0, 0, 0))
+            self.amounts.append(label)
+
+        self.load_inventory()
+        self.set_ative(self.active)
+
+    def set_ative(self, active: int):
+        self.grid[self.active].set_alpha(.25)
+        self.active = active
+        self.grid[self.active].set_alpha(.8)
+
+    def draw(self, canvas: RenderContext):
+        for i in self.grid:
+            i.draw(canvas)
+        for i in self.items:
+            i.draw(canvas)
+        for i in self.amounts:
+            i.draw(canvas)
+
+    def get_active(self):
+        if self.active >= len(self.inventory_data):
+            return []
+
+        return self.inventory_data[self.active]
+
+    def load_inventory(self):
+        resource_dir = Path('.', 'primal', 'resources').absolute()
+        try:
```

Comment:
pathlibs open context manager to check if a file exists would get you out of this try/except design. see other comment for more details.